mirascope/core/anthropic/_utils/_setup_call.py

```diff
@@ -40,14 +41,15 @@ def setup_call(
     extract: bool,
 ) -> tuple[
     Callable[..., Message] | Callable[..., Awaitable[Message]],
-    str,
+    str | None,
     list[MessageParam],
     list[type[AnthropicTool]] | None,
-    dict[str, Any],
+    AnthropicCallKwargs,
 ]:
-    prompt_template, messages, tool_types, call_kwargs = _utils.setup_call(
+    prompt_template, messages, tool_types, base_call_kwargs = _utils.setup_call(
         fn, fn_args, dynamic_config, tools, AnthropicTool, call_params
     )
+    call_kwargs: AnthropicCallKwargs = cast(AnthropicCallKwargs, base_call_kwargs)
```

Comment:
if you drop the explicit typing (i.e. `call_kwargs = cast(AnthropicCallKwargs, base_call_kwargs)`) then the below issue will be resolved and the original line will work :)

Reply:
Which error is it?
When I stop casting, I get key errors for other TypedDict-related issues.

Reply:
> I found the solution, see other comment on assignment to call_kwargs above
I see.

Reply:
yeah you still need to cast, it's just dropping `call_kwargs: AnthropicCallKwargs = ...` and instead using just `call_kwargs = ...`

Reply:
ummm, I got same error :(
```python
/Users/koudai/PycharmProjects/mirascope/mirascope/core/anthropic/_utils/_setup_call.py:78:59 - error: Expression of type "tuple[Overload[(*, max_tokens: int, messages: Iterable[MessageParam], model: ModelParam, metadata: Metadata | NotGiven = NOT_GIVEN, stop_sequences: List[str] | NotGiven = NOT_GIVEN, stream: NotGiven | Literal[False] = NOT_GIVEN, system: str | Iterable[TextBlockParam] | NotGiven = NOT_GIVEN, temperature: float | NotGiven = NOT_GIVEN, tool_choice: ToolChoice | NotGiven = NOT_GIVEN, tools: Iterable[ToolParam] | NotGiven = NOT_GIVEN, top_k: int | NotGiven = NOT_GIVEN, top_p: float | NotGiven = NOT_GIVEN, extra_headers: Headers | None = None, extra_query: Query | None = None, extra_body: Body | None = None, timeout: float | Timeout | NotGiven | None = NOT_GIVEN) -> Coroutine[Any, Any, Message], (*, max_tokens: int, messages: Iterable[MessageParam], model: ModelParam, stream: Literal[True], metadata: Metadata | NotGiven = NOT_GIVEN, stop_sequences: List[str] | NotGiven = NOT_GIVEN, system: str | Iterable[TextBlockParam] | NotGiven = NOT_GIVEN, temperature: float | NotGiven = NOT_GIVEN, tool_choice: ToolChoice | NotGiven = NOT_GIVEN, tools: Iterable[ToolParam] | NotGiven = NOT_GIVEN, top_k: int | NotGiven = NOT_GIVEN, top_p: float | NotGiven = NOT_GIVEN, extra_headers: Headers | None = None, extra_query: Query | None = None, extra_body: Body | None = None, timeout: float | Timeout | NotGiven | None = NOT_GIVEN) -> Coroutine[Any, Any, AsyncStream[RawMessageStreamEvent]], (*, max_tokens: int, messages: Iterable[MessageParam], model: ModelParam, stream: bool, metadata: Metadata | NotGiven = NOT_GIVEN, stop_sequences: List[str] | NotGiven = NOT_GIVEN, system: str | Iterable[TextBlockParam] | NotGiven = NOT_GIVEN, temperature: float | NotGiven = NOT_GIVEN, tool_choice: ToolChoice | NotGiven = NOT_GIVEN, tools: Iterable[ToolParam] | NotGiven = NOT_GIVEN, top_k: int | NotGiven = NOT_GIVEN, top_p: float | NotGiven = NOT_GIVEN, extra_headers: Headers | None = None, extra_query: Query | None = None, extra_body: Body | None = None, timeout: float | Timeout | NotGiven | None = NOT_GIVEN) -> Coroutine[Any, Any, Message | AsyncStream[RawMessageStreamEvent]]] | Overload[(*, max_tokens: int, messages: Iterable[MessageParam], model: ModelParam, metadata: Metadata | NotGiven = NOT_GIVEN, stop_sequences: List[str] | NotGiven = NOT_GIVEN, stream: NotGiven | Literal[False] = NOT_GIVEN, system: str | Iterable[TextBlockParam] | NotGiven = NOT_GIVEN, temperature: float | NotGiven = NOT_GIVEN, tool_choice: ToolChoice | NotGiven = NOT_GIVEN, tools: Iterable[ToolParam] | NotGiven = NOT_GIVEN, top_k: int | NotGiven = NOT_GIVEN, top_p: float | NotGiven = NOT_GIVEN, extra_headers: Headers | None = None, extra_query: Query | None = None, extra_body: Body | None = None, timeout: float | Timeout | NotGiven | None = NOT_GIVEN) -> Message, (*, max_tokens: int, messages: Iterable[MessageParam], model: ModelParam, stream: Literal[True], metadata: Metadata | NotGiven = NOT_GIVEN, stop_sequences: List[str] | NotGiven = NOT_GIVEN, system: str | Iterable[TextBlockParam] | NotGiven = NOT_GIVEN, temperature: float | NotGiven = NOT_GIVEN, tool_choice: ToolChoice | NotGiven = NOT_GIVEN, tools: Iterable[ToolParam] | NotGiven = NOT_GIVEN, top_k: int | NotGiven = NOT_GIVEN, top_p: float | NotGiven = NOT_GIVEN, extra_headers: Headers | None = None, extra_query: Query | None = None, extra_body: Body | None = None, timeout: float | Timeout | NotGiven | None = NOT_GIVEN) -> Stream[RawMessageStreamEvent], (*, max_tokens: int, messages: Iterable[MessageParam], model: ModelParam, stream: bool, metadata: Metadata | NotGiven = NOT_GIVEN, stop_sequences: List[str] | NotGiven = NOT_GIVEN, system: str | Iterable[TextBlockParam] | NotGiven = NOT_GIVEN, temperature: float | NotGiven = NOT_GIVEN, tool_choice: ToolChoice | NotGiven = NOT_GIVEN, tools: Iterable[ToolParam] | NotGiven = NOT_GIVEN, top_k: int | NotGiven = NOT_GIVEN, top_p: float | NotGiven = NOT_GIVEN, extra_headers: Headers | None = None, extra_query: Query | None = None, extra_body: Body | None = None, timeout: float | Timeout | NotGiven | None = NOT_GIVEN) -> (Message | Stream[RawMessageStreamEvent])], str | None, list[MessageParam], list[type[AnthropicTool]] | None, dict[str, object]]" is incompatible with return type "tuple[((...) -> Message) | ((...) -> Awaitable[Message]), str | None, list[MessageParam], list[type[AnthropicTool]] | None, AnthropicCallKwargs]"
    "dict[str, object]" is incompatible with "AnthropicCallKwargs" (reportReturnType)
1 error, 0 warnings, 0 informations 

```

Reply:
OK, I understand. `max_tokens` is required field. But, `{"model": model,  "messages": messages}` doesn't have  `max_tokens`.
It  means the lien combine `AnthropicCallKwargs ` and `dict`.

Reply:
pyright passed. 
```
call_kwargs |= {
        "model": model,
        "messages": messages,
        "max_tokens": call_kwargs["max_tokens"],
    }
```

Reply:
looks like this got resolved as desired :)